r3bbase/R3BFileSource.cxx

```diff
@@ -357,6 +375,45 @@ Bool_t R3BFileSource::Init()
         fInputFile.seekg(0, std::ios::beg);
     }
 
+    if (fInputFile.is_open())
+    {
+        LOG(INFO) << "R3BFileSource::Init() Reading RunId file";
+        std::string buffer;
+        char* p;
+        do
+        {
+            getline(fInputFile, buffer);
+            LOG(INFO) << "read from file: \"" << buffer << "\"";
+            if (buffer.find("fRunID") == 0)
+            {
+                buffer.erase(0, buffer.find('=') + 1);
+
+                frunid[array_size] = strtol(buffer.c_str(), &p, 16);
```

Comment:
It is not checked, if array_size is still within the allowed limits. Being an `Int_t`, it may as well have a negative value, or a value larger than the number of elements allocated for `frunid`.